docs/create-mysql-db-and-users.md

```diff
@@ -0,0 +1,74 @@
+# Create MySQL database and user
+
+## How to create a MySQL database/user on RDS instance
+
+You'll need to know three things:
+
+- Database name (`{Snake-cased app name}_production`, e.g. `collections_publisher_production`)
+- Database user (find this in the [config/database.yml](https://github.com/alphagov/collections-publisher/blob/e77c397cef35865aa5198f41e463f2bb7f8e688c/config/database.yml#L4) of your app. Beware that some apps - Whitehall - have multiple users to create)
+- Database user's password (you'll need to extract the value from govuk-secrets, e.g. [govuk::node::s_collections_publisher_db_admin::mysql_db_password](https://github.com/alphagov/govuk-secrets/blob/5981ef269e3e9be50a03c56e81ac530c8973d7b7/puppet_aws/hieradata/integration_credentials.yaml#L89). Make sure you look at the same environment, i.e. look in integration_credentials.yaml when making the database/user on Integration)
+
+Connect to the DB admin machine, e.g.
+
+```
+gds govuk connect ssh -e integration collections_publisher_db_admin
+```
+
+Export the three bits of data:
+
+```
+export TMP_MYSQL_DB_NAME='collections_publisher_production' ;
+export TMP_MYSQL_DB_USER='collections_pub' ;
+export TMP_MYSQL_DB_PASSWORD='THE_PASSWORD' ;
+```
+
+Create the user:
+
+```
+sudo mysql --defaults-extra-file=/root/.my.cnf -e \
+"CREATE USER IF NOT EXISTS '$TMP_MYSQL_DB_USER'@'%' IDENTIFIED WITH 'mysql_native_password' BY '${TMP_MYSQL_DB_PASSWORD}';"
```

Comment:
Note that I originally tried to use [the same command we usually use](https://github.com/alphagov/govuk-puppet/pull/11373/files#diff-b79e25034d8c9b4d13c48d6675cf5c2b1ca6a19175bbd9e691665677242828beR11), i.e. `AS` instead of `BY`:

```diff
- IDENTIFIED WITH 'mysql_native_password' AS '${TMP_MYSQL_DB_PASSWORD}';"
+ IDENTIFIED WITH 'mysql_native_password' BY '${TMP_MYSQL_DB_PASSWORD}';"
```

But this led to an error: `The password hash doesn't have the expected format.`.
I attempted to fix this by changing `TMP_MYSQL_DB_PASSWORD` for `mysql_password(TMP_MYSQL_DB_PASSWORD)` (as per [how we've done things previously](https://github.com/alphagov/govuk-puppet/pull/11373/files#diff-b79e25034d8c9b4d13c48d6675cf5c2b1ca6a19175bbd9e691665677242828beL13)) but found that I'd get `mysql_password: command not found` (even if running the process as `puppet`).

I was concerned that perhaps the password is stored unhashed in the database, but running `sudo mysql --defaults-extra-file=/root/.my.cnf -e "SELECT * FROM mysql.user WHERE user='collections_pub';"`, I can't see the original password in the output, so I don't _think_ we're losing any security in taking this approach.